spec/models/message_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+
+RSpec.describe Message, type: :model do
+  describe "Associations" do
+    it{should belong_to(:user).with_foreign_key(:user_send_id)}
+  end
+
+  describe "Scopes" do
+    let(:user1){FactoryBot.create :user}
+    let(:user2){FactoryBot.create :user}
+
+    let!(:message1) do
+      FactoryBot.create :message, user_send_id: user1.id,
+    user_receive_id: user2.id
+    end
+    let!(:message2) do
+      FactoryBot.create :message, user_send_id: user2.id,
+    user_receive_id: user1.id
+    end
+    let!(:message3) do
+      FactoryBot.create :message, user_send_id: user1.id,
+    user_receive_id: user2.id
+    end
+
+    it "newest message" do
+      Message.newest.pluck(:id).should eq [message3.id,
+                                           message2.id, message1.id]
+    end
+
+    it "search by receiver id" do
```

Comment:
describer, context và it có tác dụng gì

Reply:
- `describe` tạo ra một example group và khai báo sẽ test cái gì. Bên trong describe block có thể khai báo các nested group sử dụng `describe` hoặc `context`, hoặc các examples sử dụng `it`.
- `context` dùng để tạo nested group và khai báo sẽ test cái gì, đồng thời giúp cho tests trở nên rõ ràng, dễ đọc và có tổ chức tốt hơn.
- `it` dùng để khởi tạo example và khai báo kết quả mong muốn của test.